specs/eventing/overview.md

```diff
@@ -1,97 +1,146 @@
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+![Overview of objects](images/eventing-overview.svg)
+
+<!-- Generated from the following UML using PlantUML:
+@startuml
+' Remove shadows
+skinparam shadowing false
+' Set ordering
+left to right direction
+
+usecase Addressable <<Knative Service>>
+usecase Addressable AS A2 <<Kubernetes Service>>
+
+package "eventing.knative.dev" {
+agent Broker
+agent "Trigger" as T1
+agent "Trigger" as T2
+}
+
+Broker <-- T1 : broker
+Broker <-- T2 : broker
+Addressable <-u- T1 : subscriber
+
+package "messaging.knative.dev" {
+agent Channel
+agent "Subscription" as S1
+agent "Subscription" as S2
+}
+
+Channel <-- S1 :channel
+Channel <-- S2 :channel
+Addressable <-u- S2 :subscriber
+A2 <-u- S2 : reply
+@enduml
+-->
+
+## Interface Contracts
+
+In addition to the concrete types described below in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the referenced resources must support. The
+following interface contracts define resource fragments and partial schemas
+(required fields on an arbitrary API object) which form a basis for Knative
+Eventing.
+
+### Addressable
+
+**Addressable** resources expose a resource `address` (HTTP URL) in their
+`status` object. The URL is used as a destination for delivery of events to the
+resource; the exposed URL must implement the
+[data plane contract](data-plane.md) for receiving events.
+
+[**Broker**](#broker) and [**Channel**](#channel) both implement **Addressable**
+to receive events from other components.
+
+### Destination
+
+**Destination** is an interface (resource fragment) which is used consistently
+through Knative Eventing to reference a event delivery destination. A
+Destination eventually resolves the supplied information to a URL, and may be a
+simple URL or relative to an **Addressable** object reference; it also supports
+a Kubernetes Service object reference (as a special case). An absolute URL in a
+Destination may be used to reference cluster-external resources such as a
+virtual machine or SaaS service.
+
+### Event Source
+
+**Event Sources** are resources which generate events and may be configured to
+deliver the events to a **Destination** designated by a `sink` object in the
+resource's `spec`. The Knative Eventing spec does not define any specific event
+sources, but does define common interfaces for discovering and managing event
+sources.
+
+## Eventing
+
+### Broker
+
+**Broker** provides a central event-routing hub which exposes a URL address
+which event senders may use to submit events to the router. A Broker may be
```

Comment:
s/the router/the broker/   I know it's almost the same thing but I viewed your use of the term "event-routing hub" more like a descriptive thing and not implying the terms "broker" and "router" can be used interchangeably.